src/coprocessor/dag/rpn_expr/impl_op.rs

```diff
@@ -26,9 +26,41 @@ impl RpnFnLogicalAnd {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct RpnFnLogicalOr;
+
+impl_template_fn! { 2 arg @ RpnFnLogicalOr }
+
+impl RpnFnLogicalOr {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg0: &Option<i64>,
+        arg1: &Option<i64>,
+    ) -> Result<Option<i64>> {
+        // Rule 1. As long as there is `Some(x)` where x is not 0, the result is `1`.
+        // Rule 2. As long as there is `None`, the result is `None`.
+        // Rule 3. For the rest, the result is `0`.
+        Ok(match arg0 {
```

Comment:
It's better to use the following form:

```
Ok(match (arg0, arg1) {
    (Some(0), _) | (_, Some(0)) => Some(0),
    (None, _) | (_, None) => None,
     _ => Some(1),
})
```
Above reference to the implementation ``RpnFnLogicalAnd`

Reply:
I believe this logic is incorrect. Thinking about `arg0` is `Some(0)` and `arg1` is `Some(1)`.

Reply:
The logic is copy from `RpnFnLogicalAnd`, I mean this form not this code snippet. 😢

Reply:
How about:
```
Ok(match (arg0, arg1) {
    (Some(0), Some(0)) => Some(0),
    (None, None) | (None, Some(0)) | (Some(0), None) => None,
     _ => Some(1),
})
```